sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -510,32 +510,62 @@ class Analyzer(
    * Replaces [[UnresolvedRelation]]s with concrete relations from the catalog.
    */
   object ResolveRelations extends Rule[LogicalPlan] {
-    private def lookupTableFromCatalog(u: UnresolvedRelation): LogicalPlan = {
+
+    def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
+      case i @ InsertIntoTable(u: UnresolvedRelation, parts, child, _, _) if child.resolved =>
+        i.copy(table = EliminateSubqueryAliases(lookupTableFromCatalog(u)))
+      case u: UnresolvedRelation => resolveRelation(u)
+    }
+
+    // If the unresolved relation is running directly on files, we just return the original
+    // UnresolvedRelation, the plan will get resolved later. Else we lookup the table from catalog
+    // and change the current database name if it is a view.
+    def resolveRelation(
+        plan: LogicalPlan,
+        currentDatabase: Option[String] = None): LogicalPlan = plan match {
```

Comment:
@hvanhovell When views or tables are not resolved, our Analyzer treats them as unresolved relations. Conceptually, the views' `currentDatabase` should be part of `UnresolvedRelation`. Thus, maybe we can just move it into `UnresolvedRelation`?